pkg/capture/outputlocation/managed/storageaccount.go

```diff
@@ -0,0 +1,358 @@
+	azclients "github.com/microsoft/retina/pkg/provider/azure/clients"
+)
+
+const (
+	defaultConainerName     = "retina-capture"
+	storageAccountNamePreix = "retinacapture"
+
+	tagKeyCreatedBy = "createdBy"
+
+	// expiryDurationThreshold defines the minimal duration of the user delegation SAS expiry
+	// to let the SAS expiry be tolerable of small capture duration.
+	expiryDurationThreshold = 10 * time.Minute
+
+	// immutabilityPeriodSinceCreationInDays defines the days to protect the blob
+	// from being modified.
+	immutabilityPeriodSinceCreationInDays = 3
+
+	// durationMutiplier is the multiplier to the duration of the capture to set
+	// the expiry time of the SAS URL.
+	durationMutiplier = 2
+)
+
+func getStorageAccountTag() map[string]*string {
+	return map[string]*string{
+		tagKeyCreatedBy: to.Ptr("retina"),
+	}
+}
+
+// getStorageAccountName returns a unique storage account name.
+// Limitations of storage account name:
+// Storage account name must be between 3 and 24 characters in length and use numbers and lower-case letters only.
+func getStorageAccountName() string {
+	uniqueID := strconv.FormatInt(time.Now().Unix(), 10)
+	return storageAccountNamePreix + uniqueID
+}
+
+// StorageAccountManager manages the lifecycle of the storage account.
+type StorageAccountManager struct {
+	storageAccountName string
+
+	// uniqueContainerPerNamespace allows unique container namespace for different namespaces.
+	uniqueContainerPerNamespace bool
+
+	// containerCreated caches the created container to avoid duplicate creation.
+	// Instead of querying the storage account for existing containers, we cache
+	// the created containers to reduce the API call to Azure ARM.
+	// Because the container creation is idempotent, we don't need a lock around
+	// the map when multiple Captures  are created in parallel and enter container
+	// creation function.
+	containerCreated map[string]struct{}
+
+	azClients *azclients.AZClients
+
+	logger *log.ZapLogger
+}
+
+func NewStorageAccountManager() *StorageAccountManager {
+	return &StorageAccountManager{
+		uniqueContainerPerNamespace: true,
+		containerCreated:            make(map[string]struct{}),
+		logger:                      log.Logger().Named("storageAccount"),
+	}
+}
+
+func (sam *StorageAccountManager) Init(configFile string) error {
+	azClients, err := azclients.NewAZClients(configFile)
+	if err != nil {
+		return fmt.Errorf("failed to create Azure clients: %w", err)
+	}
+	sam.azClients = azClients
+
+	if err := sam.validateAuthConfig(); err != nil {
+		sam.logger.Error("No all configurations are set, please refer to TODO(add a link)", zap.Error(err))
+		return fmt.Errorf("failed to validate auth config: %w", err)
+	}
+
+	// All setup steps should be idempotent to withstand storage account manager restart.
+	ctx := context.Background()
+	if err := sam.SetupStorageAccount(ctx); err != nil {
+		return fmt.Errorf("failed to setup storage account: %w", err)
+	}
+
+	if !sam.uniqueContainerPerNamespace {
```

Comment:
@nitindagar0 when uniqueContainerPerNamespace is true, we'll create container for the namespace of the capture in both pod and node cases. So we are expecting the user to capture the traffic of pods on the access-allowed namespace, and for node traffic which is not scoped by namespace, we still expect the user capture the host traffic and upload it to the desired namespace and then continer.